### DISCUSSION TITLE
Deploy a smart contract locally using Forge

### DISCUSSION THREAD

**Discussion by frgpy** [Cyfrin/foundry-full-course-cu#2096] [General]
hii !! 
I am taking the course **Deploy a smart contract locally using Forge** and I encounter this error when I run this command: 
>>forge create SimpleStorage --rpc-url http://127.0.0.1:8545 --interactive
**result:** 
[⠊] Compiling...
No files changed, compilation skipped
Error: 
error sending request for url (http://127.0.0.1:8545/)

Context:
- Error #0: error sending request for url (http://127.0.0.1:8545/)
- Error #1: client error (Connect)
- Error #2: tcp connect error: Connection refused (os error 61)
- Error #3: Connection refused (os error 61)

I think the problem is that I can't add the new network to Metamask : 

new rpc url : http://127.0.0.1:8545
chain id : 1337 

"Could not fetch chain ID. Is your RPC URL correct?"

🐥 can anyone give any tip that can help please ? 🐥

**Comment by EngrPips** [comment]
Hello @frgpy, You need to run `anvil` in your terminal this will spin up a local anvil blockchain then you leave it running and open another terminal where you will run `forge create SimpleStorage --rpc-url http://127.0.0.1:8545/ --interactive`

**Reply by frgpy** [reply]
Thanks for the answer 
Yes, that's what I do on two separate terminals in Visual Studio. Should I do it on a terminal detached from VSC? 

On Metamask I can't add anvil or localhost, the problem could come from there? 
I have this message on MetaMask : "Could not fetch chain ID. Is your RPC URL correct?"

**Reply by EngrPips** [reply]
I mean you don't need to do anything with metamask to be able to run this `forge create SimpleStorage --rpc-url http://127.0.0.1:8545/ --interactive`, You  just need to have anvil running in a separate terminal then run your `forge create` command and it should work. You don't need to do it on terminal detached from `VSC` it should work in `VSC` terminal. chainid for anvil is `31337` and not `1337`

**Reply by frgpy** [reply]
Thanks a lot!  
I started all over again following your advice and it works! :)

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out. Keep crushing it.
